src/cargo-fmt/main.rs

```diff
@@ -236,9 +238,9 @@ fn get_targets(strategy: &CargoFmtStrategy) -> Result<HashSet<Target>, io::Error
 
 fn get_targets_root_only(targets: &mut HashSet<Target>) -> Result<(), io::Error> {
     let metadata = get_cargo_metadata(None)?;
-    let current_dir = env::current_dir()?.canonicalize()?;
+    let current_dir = absolute_path(env::current_dir()?)?;
     let current_dir_manifest = current_dir.join("Cargo.toml");
-    let workspace_root_path = PathBuf::from(&metadata.workspace_root).canonicalize()?;
+    let workspace_root_path = absolute_path(PathBuf::from(&metadata.workspace_root))?;
     let in_workspace_root = workspace_root_path == current_dir;
 
     for package in metadata.packages {
```

Comment:
based on the findings in https://github.com/rust-lang/rustfmt/pull/3309/files#diff-0f04fceb10cfdeda3716c7824a8dc2deR246, maybe you need to get the absolute_path of `manifest_path`:

```rust
absolute_path(PathBuf::from(&package.manifest_path)) == current_dir_manifest
```

Reply:
@topecongiro can we merge this PR with #3309 ? this would fix also @rivy 's comment https://github.com/rust-lang/rustfmt/pull/3324#issuecomment-461269425

Reply:
I think this should be a bug against `PathBuf.canonicalize()`. Any conversion to UNC must use the absolute path. PathBuf is taking that responsibility and should do it itself.

Here, as a workaround, I think you should use both, canonicalize the absolute path...

```rust
let workspace_root_path = absolute_path(PathBuf::from(&metadata.workspace_root))?.canonicalize()?;
```

Reply:
I expect a wrapper function (maybe just a passthru for non-windows OS) which executes this work-around would be a good way to proceed. And, I'd replace all uses of canonicalize() in the code base. A quick repo search shows only about about eight uses.

Not meaning to add more work, but any path comparisons should probably be reviewed for possibly needing canonicalizing of both sides. And I'm not sure where those might be scattered throughout the code base.

Reply:
review this in #3309 once this PR is merged.